src/rgw/rgw_rest_s3.cc

```diff
@@ -473,7 +482,7 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs,
     } catch (const buffer::error&) {}
   }
 
-  if (multipart_parts_count) {
+  if (multipart_parts_count && multipart_parts_count > 0) {
```

Comment:
since `multipart_parts_count` is a `std::optional<int>`, shouldn't these comparisons be `*multipart_parts_count > 0`? it's also not clear how it would end up with a 0 value given the code in rgw_rados.cc:
```c++
params.parts_count = (cur_part_id == 1) ? 1 : cur_part_id - 1;;
```

Reply:
that's true but this is defensive programming that tracks the structure of the variable

Reply:
yes, I misread